### PR TITLE
T14417 filter closure

### PR DESCRIPTION
**Pull request by niden**
Hello!

* Type: bug fix
* Link to issue: #14417 

**In raising this pull request, I confirm the following:**

- [x] I have read and understood the [Contributing Guidelines](https://github.com/phalcon/cphalcon/blob/master/CONTRIBUTING.md)
- [x] I have checked that another pull request for this purpose does not exist
- [x] I wrote some tests for this PR
- [x] I have updated the relevant CHANGELOG
- [ ] I have created a PR for the [documentation](https://github.com/phalcon/docs) about this change

Fixed `Phalcon\Filter` to correctly work with closures.

Thanks

### CODE REVIEW

**Comment by niden** [comment]
cc @ruudboon @sergeyklay

**Review by ruudboon — DISMISSED** [review]
Thnx @niden for the fast fix and cleaning up the Changelog. Looks good to me.

File: phalcon/Filter.zep [hunk 1]
```diff
@@ -77,9 +77,7 @@ class Filter implements FilterInterface
         if !isset(this->services[name]) {
             let definition = this->mapper[name];
             if typeof definition === "string" {
-                let this->services[name] = create_instance(definition);
-            } elseif definition instanceof Closure {
-                let this->services[name] = call_user_func(definition);
+                let this->services[name] = new {definition}();
```
Comment (sergeyklay) [inline phalcon/Filter.zep]:
```suggestion
                let this->services[name] = create_instance(definition);
```

**Review by sergeyklay — DISMISSED** [review]
A small correction. But in general LGTM

**Review by Jeckerson — APPROVED** [review]

**Review by ruudboon — APPROVED** [review]

**Pull request merged**